scripts/check_deployment_secrets.py

```diff
@@ -0,0 +1,155 @@
+            return
+        if not (('installed' not in secret) or ('client_email' not in secret)) and (self.stage in self.stages):
+            self.malformed_secrets.append(secret_name)
+            return
+        return secret
+
+    def fetch_terraform_output(self, output_name, output_infra_dir):
+        """See: https://www.terraform.io/docs/commands/output.html"""
+        output_infra_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'infra', output_infra_dir))
+
+        # populate infra's vars for the current stage
+        self.run_cmd(cmd=f'make -C infra')
+        self.run_cmd(cmd=f'terraform refresh', cwd=output_infra_dir)
+
+        # query terraform as to what the needed var is and return it
+        terraform_output = self.run_cmd(cmd=f'terraform output {output_name}', cwd=output_infra_dir)
+        return terraform_output.strip()
+
+    def check(self, current, expected, secret):
+        if current not in expected:
+            self.incomplete_secrets.append({'secret': secret,
+                                            'current': current,
+                                            'expected': expected})
+
+    def run(self):
+        # do not check user-custom deploys
+        if self.stage in self.stages:
+            print(f'Now checking the secrets for {self.stage}...')
+            self.check(self.app_secret['installed']['auth_uri'], self.auth_uri, secret=self.app_secret_name)
+            self.check(self.app_secret['installed']['token_uri'], self.token_uri, secret=self.app_secret_name)
+            self.check(self.gcp_cred_secret['type'], self.type, secret=self.gcp_cred_secret_name)
+            self.check(self.gcp_cred_secret['project_id'], self.project, secret=self.gcp_cred_secret_name)
+            self.check(self.gcp_cred_secret['client_email'], self.email, secret=self.gcp_cred_secret_name)
+            print(f'Secret check complete for {self.stage}.')
+
+        if self.missing_secrets or self.incomplete_secrets or self.malformed_secrets:
+            for s in self.incomplete_secrets:
+                self.error_message += f'\n' \
+                                      f'The following secret                : {s["secret"]}\n' \
+                                      f'Had the unexpected setting          : {s["current"]}\n' \
+                                      f'When one of these items was expected: {s["expected"]}\n'
+            self.error_message += '\n'
+            for s in self.missing_secrets:
+                self.error_message += f'The following secret was missing    : {s}\n'
+            self.error_message += '\n'
+            for s in self.malformed_secrets:
+                self.error_message += f'The following secret seems malformed: {s}\n'
+            raise ValueError(self.error_message)
+
+
+def main(stage=None):
+    if len(sys.argv) > 1:
+        stage = sys.argv[1]
+    elif not stage:
+        stage = os.environ['DSS_DEPLOYMENT_STAGE']
+    s = SecretsChecker(stage)
```

Comment:
perhaps should ensure that `if stage in STAGES` where `STAGES` are  `['dev','integration','staging','prod']` just to make sure the `DSS_DEPLOYMENT_STAGE` is not set to something arbitrary.

Reply:
It does this within the class itself.